File: Back-End/ProdeinWebApi/Application/Common/Entities/Usuario.cs
using System;
using System.Collections.Generic;
using System.Linq;
using System.Web;

namespace ProdeinWebApi.Application.Common.Entities
{
    public class Usuario : Entidad
    {
        private string _nombre;
        private string _apellido;
        private string _password;
        private string _rol;
        private DateTime _fechaCreacion;
        private DateTime _fechaModificacion;
        private string _direccionIP;
        private string _token;

        public string Nombre { get => _nombre; set => _nombre = value; }
        public string Apellido { get => _apellido; set => _apellido = value; }
        public string Password { get => _password; set => _password = value; }
        public string Rol { get => _rol; set => _rol = value; }
        public DateTime FechaCreacion { get => _fechaCreacion; set => _fechaCreacion = value; }
        public DateTime FechaModificacion { get => _fechaModificacion; set => _fechaModificacion = value; }
        public string DireccionIP { get => _direccionIP; set => _direccionIP = value; }
        public string Token { get => _token; set => _token = value; }
    }
}

File: Front-End/ProdeinWebApp/src/app/common/models/usuario.ts
import { BaseModel } from "./base";

export class Usuario extends BaseModel
{
    public nombreLogin?: string;
    public nombre?: string;
    public apellido?: string;
    public password?: string;
    public rol?: string;
    public fechaCreacion?: string;
    public fechaModificacion?: string;
    public direccionIP?: string;
    public token?: string;

    public static clone( usuario: Usuario ): Usuario
    {
        let result : Usuario = new Usuario() 
       
        result.id = usuario.id;
        result.nombreLogin = usuario.nombreLogin;
        result.nombre = usuario.nombre;
        result.apellido = usuario.apellido;
        result.password = usuario.password;
        result.rol = usuario.rol;
        result.token = usuario.token;
        result.fechaCreacion = usuario.fechaCreacion;
        result.fechaModificacion = usuario.fechaModificacion;
        result.direccionIP = usuario.direccionIP;

        return result;
    }
}
